src/process/mod.rs

```diff
@@ -388,6 +510,54 @@ pub struct MemoryMap {
     pub pathname: MMapPath,
 }
 
+impl MemoryMap {
+    /// Used internally in Process::smaps() as a "default value" thing
+    fn new() -> Self {
+        Self {
+            address: (0, 0),
+            perms: "".into(),
+            offset: 0,
+            dev: (0, 0),
+            inode: 0,
+            pathname: MMapPath::new(),
+        }
+    }
+    fn from_line(line: &str) -> ProcResult<MemoryMap> {
+        let mut s = line.splitn(6, ' ');
+        let address = expect!(s.next());
+        let perms = expect!(s.next());
+        let offset = expect!(s.next());
+        let dev = expect!(s.next());
+        let inode = expect!(s.next());
+        let path = expect!(s.next());
+
+        Ok(MemoryMap {
+            address: split_into_num(address, '-', 16)?,
+            perms: perms.to_string(),
+            offset: from_str!(u64, offset, 16),
+            dev: split_into_num(dev, ':', 16)?,
+            inode: from_str!(u64, inode),
+            pathname: MMapPath::from(path)?,
+        })
+    }
+}
+
+/// Represents the information about a specific mapping as presented in /proc/<pid>/smaps
+///
+/// To construct this structure, see [Process::smaps()]
+#[derive(Default)]
+pub struct MemoryMapData {
+    /// Key-Value pairs that may represent statistics about memory usage, or other interesting things,
+    /// such a "ProtectionKey"(if you're on X86 and that kernel config option was specified).
+    ///
+    /// Check your manpage for more information
+    pub key_value_pairs: HashMap<String, u64>,
```

Comment:
I don't like this name, however I am not sure what a better name would be. Any suggestions?

Reply:
What about just calling this `map`  (since it is a HashMap after all)?   Or maybe `data`?

Reply:
I went with `map` since I liked that name more, but I don't have a particularly strong opinion on this.